Feature/update image upload 40768943

CrazyOptimist opened the pull request:
* update image uploading modal (Figma 7.3 - 7.5)
https://uistorybook.pastel.network/?path=/story/features-add-nft--add-nft
* add commit id to smartcrop.js dep
* fix storybook build error (https://github.com/babel/babel/issues/11622)
* add Tooltip2 common component because Tooltip already exists but not customizable enough
* change optimization slider to use % quality instead of MB because letting user choose size doesn't seem to be implementable

ngvtuan reviewed (commented):
<img width="1182" alt="Can not close the popup" src="https://user-images.githubusercontent.com/10944387/125015631-0fd2f100-e09a-11eb-974d-c9603ce7f2bf.png">
![zoom is not same with the design](https://user-images.githubusercontent.com/10944387/125015653-1c574980-e09a-11eb-85e7-4ae9d83a1898.jpg)

ngvtuan reviewed (commented):
![ui issue 1](https://user-images.githubusercontent.com/10944387/126601312-7ebb0db1-ac47-45da-9f61-2f0dd0097844.jpg)

![ui issue 2](https://user-images.githubusercontent.com/10944387/126601327-9486e7c2-2e58-469b-9198-26a68b7ec8b9.jpg)

![ui issue 3](https://user-images.githubusercontent.com/10944387/126601329-fb937567-6d5d-4601-b746-9c3afc8e01c5.jpg)

goal-shevchenko reviewed (changes requested):
<img width="1088" alt="Screen Shot 2021-07-22 at 1 15 27 PM" src="https://user-images.githubusercontent.com/63484644/126624757-7f233e90-7381-446a-a0db-295a55e4bb4e.png">


![image](https://user-images.githubusercontent.com/63484644/126627001-f3b5451d-5c2f-48d5-8b78-664d4235f244.png)

CrazyOptimist commented on src/common/components/Tooltip2/Tooltip2.module.css:
```diff
@@ -0,0 +1,8 @@
+.arrowDown {
```
tailwind can't handle different coloring different sides of a border

CrazyOptimist commented:
I needed to merge PR of image optimization to avoid further conflicts

Removed extra close button on cropping and in full screen

Changed magnification zoom level significantly because with small zoom level it's impossible to see the difference, so now has around 25 times zoom depending on image size

Before/After now has updated design with white background

Removed custom cursor icon from previous design, now it is a standard cursor.

Please ignore other screens, this PR is for "preview" modal with optimization slider.

ngvtuan reviewed (commented):
This PR is ok to merge. But please resolve the conflict.

goal-shevchenko reviewed (commented):
Looks good to me then except for merge conflict.

CrazyOptimist commented on src/features/nft/addNFT/previewStep/PreviewStepModal.tsx:
```diff
@@ -0,0 +1,247 @@
+        }
+      } catch (error) {
+        setOptimizingState('fail')
+        console.error(error)
```
And how to log error?

ngvtuan replied:
I think you can use https://www.npmjs.com/package/electron-log if you want to log the error into the production. If this console.log only use for development I think it doesn't need.
